spec/models/order_item_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe OrderItem, type: :model do
+  describe 'validations' do
+    it 'is valid with valid attributes' do
+      order_item = build(:order_item)
+      expect(order_item).to be_valid
+    end
+
+    it 'is not valid without an order_id' do
+      order_item = build(:order_item, order: nil)
+      expect(order_item).to_not be_valid
+      expect(order_item.errors[:order]).to include('must exist')
+    end
+
+    it 'is not valid without a product_id' do
+      order_item = build(:order_item, product: nil)
+      expect(order_item).to_not be_valid
+      expect(order_item.errors[:product]).to include('must exist')
+    end
+
+    it 'is not valid with a non-unique product within the same order' do
+      order = create(:order)
+      product = create(:product)
+      create(:order_item, order:, product:, quantity: 2)
+      order_item = build(:order_item, order:, product:, quantity: 3)
+      expect(order_item).to_not be_valid
+      expect(order_item.errors[:product_id]).to include('has already been taken')
+    end
+
+    it 'is not valid without a quantity' do
+      order_item = build(:order_item, quantity: nil)
+      expect(order_item).to_not be_valid
+      expect(order_item.errors[:quantity]).to include("can't be blank")
```

Comment:
use same convention in all code for string bhai, I would suggest to go for single quote in these strings bhai

Reply:
Okay bhai, thank you.

Reply:
Actually bhai, RuboCop formatting decided to do that because of the word `can't`, as it already have single quote.

Reply:
@shr-wtag may be we can use ('can\'t be blank') but please ask Hasib bhai to know which one would be better

Reply:
Okay bhai.